pkg/client/testclient/fixture.go

```diff
@@ -216,3 +218,46 @@ func (o objects) Add(obj runtime.Object) error {
 
 	return nil
 }
+
+func (o objects) Delete(obj runtime.Object) error {
```

Comment:
I just tested client.Pods(ns).Get("any") and the first return was podA, call it again and the second return was podB.  I understand now what you mean by sequence.

also, i will refactor my unit test to use mocks and narrow interfaces more like #6871.  Don't need this PR.